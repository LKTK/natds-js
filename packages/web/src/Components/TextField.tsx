import React, { FunctionComponent } from 'react';
import styled from 'styled-components';

import { withTheme } from '@material-ui/styles';
import { IThemeWeb } from 'Themes';
import { tokens } from '@naturacosmeticos/natds-styles';

interface ITextFieldProps {
  [propName: string]: any;
  /**
   * The input id property
   */
  id: string;
  /**
   * @optional
   */
  theme?: IThemeWeb | unknown;
  /**
   * @optional
   * Help text placed underneath the text TextField
   */
  label?: string;
  /**
   * @optional
   * Help text placed underneath the text TextField
   */
  helpText?: string;
  /**
   * @optional
   * If this field is required or not
   */
  required?: boolean;
  /**
   * @optional
   * Disables the TextField interaction
   */
  disabled?: boolean;
  /**
   * @optional
   * Optional status for component variant
   */
  status?: 'error' | 'success' | undefined;
  /**
   * @optional
   * The input type property
   */
  type?: 'text' | 'password' | 'search' | string;
  /**
   * @optional
   * Uses a textarea as input
   */
  multiline?: boolean;
}

const TextField: FunctionComponent<ITextFieldProps> = (props: ITextFieldProps) => {
  const {
    id,
    label,
    helpText,
    theme,
    required = false,
    disabled = false,
    status,
    multiline,
    placeholder = ' ',
    ...rest
  } = props;

  const content = required ? `${label} *` : label;

  return (
    <Container theme={theme}>
      <Label
        theme={theme}
        htmlFor={id}
        status={status}
        disabled={disabled}>
        {content}
      </Label>
      <Field
        theme={theme}
        id={id}
        status={status}
        disabled={disabled}
        as={multiline ? 'textarea' : 'input'}
        placeholder={placeholder}
        {...rest}
      />
      <HelpText
        theme={theme}
        status={status}
        disabled={disabled}>
        {helpText}
      </HelpText>
    </Container>
  );
};

export default withTheme(TextField);

const stateColors = {
  disabled: { type: 'text', key: 'disabled' },
  error: { type: 'error', key: 'main' },
  success: { type: 'success', key: 'main' },
  initial: { type: 'text', key: 'hint' },
  primary: { type: 'primary', key: 'main' },
};

function getProp(namespace: string, type: string, key?: string) {
  return ({ theme }: { theme: IThemeWeb }) => {
    const propNamespace = theme[namespace] || {};
    const propType = propNamespace[type] || {};

    return key ? propType[key] : propType;
  };
}

function getColorByState(initial: { type: string, key: string }) {
  return ({ theme, disabled, status = '' }: { theme: IThemeWeb, disabled: boolean, status: string | undefined }) => {
    const { type, key } = (disabled ? stateColors.disabled : stateColors[status]) || initial;

    return getProp('palette', type, key)({ theme });
  };
}

const Container = styled.div`
  display: flex;
  flex-flow: column nowrap;
  font-family: ${getProp('typography', 'fontFamily')};
`;

const Label = styled.label`
  font-size: ${getProp('typography', 'subtitle2', 'fontSize')};
  font-weight: ${getProp('typography', 'subtitle2', 'fontWeight')};
  color: ${getColorByState(stateColors.initial)};
  line-height: 1.2;
  padding: 0 0 ${tokens.spacing.spacingMicro}px;
`;

const Field = styled.input`
  border: 0;
  border-radius: ${getProp('shape', 'borderRadius')}px;
  box-shadow: ${getColorByState(stateColors.initial)} 0 0 0 1px;
  box-sizing: border-box;
  width: 100%;
  line-height: ${tokens.spacing.spacingStandard}px;
  min-height: ${tokens.spacing.spacingLarge}px;
  outline: none;
  color: ${getProp('palette', 'text', 'primary')};
  font-size: ${getProp('typography', 'body2', 'fontSize')};
  font-weight: ${getProp('typography', 'body2', 'fontWeight')};
  flex: 1 1 100%;
  padding: ${tokens.spacing.spacingSmall}px;
  resize: vertical;

  &:hover {
    box-shadow: ${getProp('palette', 'grey', '600')} 0 0 0 1px;
  }

  &:focus {
    box-shadow: ${getColorByState(stateColors.primary)} 0 0 0 2px;
  }

  &:disabled,
  &:disabled::placeholder,
  &:disabled:hover {
    color: ${getProp('palette', 'text', 'disabled')};
  }

  &:disabled,
  &:disabled:hover {
    box-shadow: ${getProp('palette', 'text', 'disabled')} 0 0 0 1px;
  }

  &::placeholder {
    color: ${getProp('palette', 'text', 'hint')};
  }

  &:placeholder-shown {
    box-shadow: ${getColorByState(stateColors.initial)} 0 0 0 1px;
  }

  &:not(:placeholder-shown) {
    box-shadow: ${getProp('palette', 'grey', '600')} 0 0 0 1px;
  }
`;

const HelpText = styled.span`
  font-size: ${getProp('typography', 'caption', 'fontSize')};
  font-weight: ${getProp('typography', 'caption', 'fontWeight')};
  color: ${getColorByState(stateColors.initial)};
  line-height: 1.2;
  padding: ${tokens.spacing.spacingMicro}px 0 0;
`;
