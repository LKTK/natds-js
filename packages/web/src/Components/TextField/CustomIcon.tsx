import React, { FunctionComponent, ComponentType } from 'react';
import styled from 'styled-components';
import { IThemeWeb } from 'Themes';
import { tokens } from '@naturacosmeticos/natds-styles';
import { getProp } from './shared';

interface ICustomIcon {
  theme: IThemeWeb | unknown;
  icon?: ComponentType | undefined;
  onIconPress?: () => (void);
}

const CustomIcon: FunctionComponent<ICustomIcon> = (props: ICustomIcon) => {
  if (!props.icon) return null;

  const { theme, icon, onIconPress } = props;
  const style = onIconPress ? { cursor: 'pointer' } : {};

  return (
    <ClearIconStyled
      theme={theme}
      style={style}
      onClick={onIconPress}>
      {icon}
    </ClearIconStyled>
  );
};

export default CustomIcon;

const ClearIconStyled = styled.i`
  width: ${tokens.spacing.spacingStandard}px;
  height: ${tokens.spacing.spacingStandard}px;
  fill:  ${getProp('palette', 'text', 'primary')};
  position: absolute;
  top: 0;
  bottom: 0;
  right: ${tokens.spacing.spacingSmall}px;
  margin: auto;
`;
