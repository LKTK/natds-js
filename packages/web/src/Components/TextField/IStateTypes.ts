export interface IStateTypes {
  type: string;
  key: string;
  borderWidth: string;
}
