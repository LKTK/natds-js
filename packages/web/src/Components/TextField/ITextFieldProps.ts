import { ComponentType } from "react";
import { IThemeWeb } from "../../Themes";

export interface ITextFieldProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [propName: string]: any;

  /**
   * The input id property
   */
  id: string;

  /**
   * @optional
   */
  theme?: IThemeWeb | unknown;

  /**
   * @optional
   * Help text placed underneath the text TextField
   */
  label?: string;

  /**
   * @optional
   * Help text placed underneath the text TextField
   */
  helpText?: string;

  /**
   * @optional
   * If this field is required or not
   */
  required?: boolean;

  /**
   * @optional
   * Disables the TextField interaction
   */
  disabled?: boolean;

  /**
   * @optional
   * Optional state for component variant
   */
  state?: "error" | "success" | undefined;

  /**
   * @optional
   * The input type property
   */
  type: string;

  /**
   * @optional
   * Uses a textarea as input
   */
  multiline?: boolean;

  /**
   * @optional
   * Mask format. Based on https://github.com/sanniassin/react-input-mask
   */
  mask?: () => (void) | string[];

  /**
   * @optional
   * Icon component
   */
  icon?: ComponentType;

  /**
   * @optional
   * Icon component callback
   */
  onIconPress?: () => void;

  /**
   * @optional
   * Icon component to use with the Search field
   */
  searchIcon?: ComponentType;

  /**
   * @optional
   * Icon component to use with the Search field
   */
  showPasswordIcon?: ComponentType;

  /**
   * @optional
   * Icon component to use with the Search field
   */
  hidePasswordIcon?: ComponentType;
}
