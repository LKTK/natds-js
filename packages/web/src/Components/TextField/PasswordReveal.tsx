import React, { FunctionComponent, Dispatch, SetStateAction, ComponentType } from 'react';
import styled from 'styled-components';
import VisibilityIcon from '@material-ui/icons/VisibilityOutlined';
import VisibilityIconOff from '@material-ui/icons/VisibilityOffOutlined';

import { IThemeWeb } from 'Themes';
import { tokens } from '@naturacosmeticos/natds-styles';
import { getProp } from './shared';

interface IPasswordReveal {
  theme: IThemeWeb | unknown;
  showing: boolean;
  onTogglePasswordReveal: Dispatch<SetStateAction<boolean>>;
  showPasswordIcon?: ComponentType;
  hidePasswordIcon?: ComponentType;
}

const baseIconStyle = `
  width: ${tokens.spacing.spacingStandard}px;
  height: ${tokens.spacing.spacingStandard}px;
  fill:  ${getProp('palette', 'text', 'primary')};
  position: absolute;
  top: 0;
  bottom: 0;
  right: ${tokens.spacing.spacingSmall}px;
  margin: auto;
  cursor: pointer;
`;

const PasswordReveal: FunctionComponent<IPasswordReveal> = (props: IPasswordReveal) => {
  const {
    theme,
    onTogglePasswordReveal,
    showing,
    showPasswordIcon = VisibilityIcon,
    hidePasswordIcon = VisibilityIconOff
  } = props;

  const Component: ComponentType<any> = showing
    ? styled(hidePasswordIcon)`${baseIconStyle}`
    : styled(showPasswordIcon)`${baseIconStyle}`;

  return (
    <Component
      theme={theme}
      onClick={() => onTogglePasswordReveal(!showing)}
    />
  );
};

export default PasswordReveal;

