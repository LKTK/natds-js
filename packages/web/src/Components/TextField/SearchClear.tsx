import React, { FunctionComponent, Dispatch, SetStateAction, ComponentType } from 'react';
import styled from 'styled-components';
import ClearRounded from '@material-ui/icons/ClearRounded';

import { IThemeWeb } from 'Themes';
import { tokens } from '@naturacosmeticos/natds-styles';
import { getProp } from './shared';

interface ISearchClear {
  theme: IThemeWeb | unknown;
  onClearSearch: Dispatch<SetStateAction<string>>;
  searchIcon?: ComponentType;
}

const SearchIconStyles = `
  width: ${tokens.spacing.spacingStandard}px;
  height: ${tokens.spacing.spacingStandard}px;
  fill:  ${getProp('palette', 'text', 'primary')};
  position: absolute;
  top: 0;
  bottom: 0;
  right: ${tokens.spacing.spacingSmall}px;
  margin: auto;
  cursor: pointer;
`;

const SearchClear: FunctionComponent<ISearchClear> = (props: ISearchClear) => {
  const { theme, onClearSearch, searchIcon = ClearRounded } = props;
  const Component: ComponentType<any> = styled(searchIcon)`${SearchIconStyles}`;

  return (
    <Component
      theme={theme}
      onClick={() => onClearSearch("")}
    />
  );
};

export default SearchClear;
