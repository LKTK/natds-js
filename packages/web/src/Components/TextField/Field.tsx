import React, { FunctionComponent, useState } from 'react';
import styled from 'styled-components';
import MaskedInput from 'react-text-mask';
import { tokens } from '@naturacosmeticos/natds-styles';

import { ITextFieldProps, getProp, getBorderByState, stateStyles } from './shared';
import PasswordReveal from './PasswordReveal';
import SearchClear from './SearchClear';
import CustomIcon from './CustomIcon';

export const TEXT_TYPE = 'text';
export const SEARCH_TYPE = 'search';
export const PASSWORD_TYPE = 'password';

const Field: FunctionComponent<ITextFieldProps> = (props: ITextFieldProps) => {
  const {
    id,
    theme,
    required = false,
    disabled = false,
    state,
    multiline,
    mask,
    placeholder = '&nbsp;',
    type,
    onIconPress,
    icon,
    searchIcon,
    showPasswordIcon,
    hidePasswordIcon,
    ...rest
  } = props;

  const fieldType = multiline ? 'textarea' : (mask ? MaskedInput : 'input');
  const [showing, togglePasswordReveal] = useState(false);
  const [value, setValue] = useState("");
  const customType = showing ? TEXT_TYPE : type;
  const showPasswordReveal = type === PASSWORD_TYPE && !icon;
  const showSearchClear = type === SEARCH_TYPE && !icon;

  return (
    <FieldContainer>
      <FieldComponent
        theme={theme}
        id={id}
        type={customType || TEXT_TYPE}
        placeholder={placeholder}
        state={state}
        disabled={disabled}
        required={required}
        as={fieldType}
        value={value}
        mask={mask}
        onChange={(event: any) => setValue(event.target.value)}
        {...rest}
      />
      {showPasswordReveal && <PasswordReveal
        theme={theme}
        showing={showing}
        onTogglePasswordReveal={togglePasswordReveal}
        showPasswordIcon={showPasswordIcon}
        hidePasswordIcon={hidePasswordIcon}
      />}
      {showSearchClear && <SearchClear
        theme={theme}
        onClearSearch={setValue}
        searchIcon={searchIcon}
      />}
      <CustomIcon
        theme={theme}
        icon={icon}
        onIconPress={onIconPress}
      />
    </FieldContainer>
  );
};

export default Field;

const FieldContainer = styled.div`
  display: flex;
  flex-flow: column nowrap;
  position: relative;
`;

export const FieldComponent = styled.input`
  border: 0;
  border-radius: ${getProp('shape', 'borderRadius')}px;
  box-sizing: border-box;
  width: 100%;
  line-height: ${tokens.spacing.spacingStandard}px;
  min-height: ${tokens.spacing.spacingLarge}px;
  outline: none;
  color: ${getProp('palette', 'text', 'primary')};
  font-size: ${getProp('typography', 'body2', 'fontSize')};
  font-weight: ${getProp('typography', 'body2', 'fontWeight')};
  flex: 1 1 100%;
  padding: ${tokens.spacing.spacingSmall}px;
  resize: vertical;

  &:disabled,
  &:disabled:hover {
    color: ${getProp('palette', 'text', 'disabled')};
    box-shadow: ${getProp('palette', 'text', 'disabled')} 0 0 0 1px;
  }

  &::placeholder {
    color: ${getProp('palette', 'text', 'hint')};
  }

  &:disabled::placeholder {
    color: ${getProp('palette', 'text', 'disabled')};
  }

  &:placeholder-shown {
    box-shadow: ${getBorderByState(stateStyles.default)};
  }

  &:not(:placeholder-shown) {
    box-shadow: ${getBorderByState(stateStyles.filled)};
  }

  &:hover:not(:read-only) {
    box-shadow: ${getBorderByState(stateStyles.hover)};
  }

  &:focus:not(:read-only) {
    box-shadow: ${getBorderByState(stateStyles.focus)};
  }
`;
