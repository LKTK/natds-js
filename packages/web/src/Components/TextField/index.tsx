import React, { FunctionComponent, forwardRef } from "react";
import { withTheme } from "@material-ui/styles";

import { ITextFieldProps as TextFieldProps } from "./shared";

import Field from "./Field";
import InputStateHelpTextProvider, { IInputStateHelpTextProviderProps } from "../InputStateHelpTextProvider";

export type ITextFieldProps = TextFieldProps & IInputStateHelpTextProviderProps;

type ITextFieldPropsWithoutTheme = Omit<ITextFieldProps, "theme">;

export const TextField: FunctionComponent<ITextFieldPropsWithoutTheme> = forwardRef((props: ITextFieldProps, ref: any) => <InputStateHelpTextProvider {...props}>
  <Field {...props} ref={ref} />
</InputStateHelpTextProvider>);

export default withTheme(TextField);
