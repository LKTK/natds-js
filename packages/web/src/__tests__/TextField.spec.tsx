import "jest-styled-components";

import * as React from "react";
import * as renderer from "react-test-renderer";

import Favorite from "@material-ui/icons/Favorite";

import { expect } from "chai";
import { shallow } from "enzyme";
import { themes } from "@naturacosmeticos/natds-styles";
import TextField from "../Components/TextField";

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const getProps = (props: any = {}) => {

  const {
    disabled,
    helpText = "Assistive text",
    id = "field",
    label = "Label",
    mask,
    placeholder = "Placeholder",
    required,
    state,
    theme = themes.natura.light,
  } = props;

  return {
    disabled,
    helpText,
    id,
    label,
    mask,
    placeholder,
    required,
    state,
    theme,
  };
};

describe("TextField component", () => {
  describe("rendering state", () => {
    test("should match snapshot - TextField (default)", () => {
      const component = renderer.create(<TextField {...getProps()} />).toJSON();

      expect(component).to.matchSnapshot("TextField default snapshot");
    });

    test("should match snapshot - TextField (readonly explicitly set to true)", () => {
      const component = renderer.create(<TextField {...getProps()} readOnly={true} />).toJSON();

      expect(component).to.matchSnapshot("TextField readonly explicitly set to true snapshot");
    });

    test("should match snapshot - TextField (readonly explicitly set to false)", () => {
      const component = renderer.create(<TextField {...getProps()} readOnly={false} />).toJSON();

      expect(component).to.matchSnapshot("TextField readonly explicitly set to false snapshot");
    });

    test("should match snapshot - TextField (default without placeholder)", () => {
      const props = getProps();

      Reflect.deleteProperty(props, "placeholder");

      const component = renderer.create(<TextField {...props} />).toJSON();

      expect(component).to.matchSnapshot("TextField default without placeholder snapshot");
    });

    test("should match snapshot - TextField (default hover)", () => {
      const component = shallow(<TextField {...getProps()} />);

      component.find("#field").simulate("mouseover");

      expect(component).to.matchSnapshot("TextField default hover snapshot");
    });

    test("should match snapshot - TextField (default focus)", () => {
      const component = shallow(<TextField {...getProps()} />);

      component.find("#field").simulate("focus");

      expect(component).to.matchSnapshot("TextField default focus snapshot");
    });

    test("should match snapshot - TextField (filled)", () => {
      const component = renderer.create(<TextField {...getProps()} value="some value" />).toJSON();

      expect(component).to.matchSnapshot("TextField filled snapshot");
    });

    test("should match snapshot - TextField (filled hover)", () => {
      const component = shallow(<TextField {...getProps()} value="some value" />);

      component.find("#field").simulate("mouseover");

      expect(component).to.matchSnapshot("TextField filled hover snapshot");
    });

    test("should match snapshot - TextField (filled focus)", () => {
      const component = shallow(<TextField {...getProps()} value="some value" />);

      component.find("#field").simulate("focus");

      expect(component).to.matchSnapshot("TextField filled focus snapshot");
    });

    test("should match snapshot - TextField (success)", () => {
      const component = renderer.create(<TextField {...getProps({ state: "success" })} />).toJSON();

      expect(component).to.matchSnapshot("TextField success snapshot");
    });

    test("should match snapshot - TextField (success hover)", () => {
      const component = shallow(<TextField {...getProps({ state: "success" })} />);

      component.find("#field").simulate("mouseover");

      expect(component).to.matchSnapshot("TextField success hover snapshot");
    });

    test("should match snapshot - TextField (success focus)", () => {
      const component = shallow(<TextField {...getProps({ state: "success" })} />);

      component.find("#field").simulate("focus");

      expect(component).to.matchSnapshot("TextField success focus snapshot");
    });

    test("should match snapshot - TextField (error)", () => {
      const component = renderer.create(<TextField {...getProps({ state: "error" })} />).toJSON();

      expect(component).to.matchSnapshot("TextField error snapshot");
    });

    test("should match snapshot - TextField (error hover)", () => {
      const component = shallow(<TextField {...getProps({ state: "error" })} />);

      component.find("#field").simulate("mouseover");

      expect(component).to.matchSnapshot("TextField error hover snapshot");
    });

    test("should match snapshot - TextField (error focus)", () => {
      const component = shallow(<TextField {...getProps({ state: "error" })} />);

      component.find("#field").simulate("focus");

      expect(component).to.matchSnapshot("TextField error focus snapshot");
    });

    test("should match snapshot - TextField (required)", () => {
      const component = renderer.create(<TextField {...getProps({ required: true })} />).toJSON();

      expect(component).to.matchSnapshot("TextField required snapshot");
    });

    test("should match snapshot - TextField (required without label)", () => {
      const component = renderer.create(<TextField {...getProps({
        label: null,
        required: true,
      })} />).toJSON();

      expect(component).to.matchSnapshot("TextField required without label snapshot");
    });

    test("should match snapshot - TextField (disabled)", () => {
      const component = renderer.create(<TextField {...getProps({ disabled: true })} />).toJSON();

      expect(component).to.matchSnapshot("TextField disabled snapshot");
    });

    test("should match snapshot - TextField (disabled hover)", () => {
      const component = shallow(<TextField {...getProps({ disabled: true })} />);

      component.find("#field").simulate("mouseover");

      expect(component).to.matchSnapshot("TextField disabled hover snapshot");
    });

    test("should match snapshot - TextField (disabled focus)", () => {
      const component = shallow(<TextField {...getProps({ disabled: true })} />);

      component.find("#field").simulate("focus");

      expect(component).to.matchSnapshot("TextField disabled focus snapshot");
    });

    test("should match snapshot - TextField (multiline)", () => {
      const component = renderer.create(<TextField {...getProps()} multiline />).toJSON();

      expect(component).to.matchSnapshot("TextField multiline snapshot");
    });

    test("should match snapshot - TextField (custom icon)", () => {
      const component = renderer.create(<TextField {...getProps()} icon={<Favorite />} />).toJSON();

      expect(component).to.matchSnapshot("TextField custom icon snapshot");
    });

    test("should match snapshot - TextField (custom icon with action)", () => {
      const component = renderer.create(<TextField {...getProps()} icon={<Favorite />} onIconPress={() => ({})} />).toJSON();

      expect(component).to.matchSnapshot("TextField custom icon with action snapshot");
    });

    test("should match snapshot - TextField (mask)", () => {
      const mask = [
        "(",
        /[1-9]/,
        /\d/,
        ")",
        " ",
        /\d/,
        /\d/,
        /\d/,
        /\d/,
        "-",
        /\d/,
        /\d/,
        /\d/,
        /\d/,
      ];
      const component = renderer.create(<TextField {...getProps({ mask })} />).toJSON();

      expect(component).to.matchSnapshot("TextField mask snapshot");
    });

    test("should match snapshot - TextField (theme change)", () => {
      const component = renderer.create(<TextField {...getProps({ theme: themes.natura.dark })} />).toJSON();

      expect(component).to.matchSnapshot("TextField theme change snapshot");
    });

    test("should match snapshot - TextField (property forwarded)", () => {
      const component = renderer.create(<TextField {...getProps()} type="email" />).toJSON();

      expect(component).to.matchSnapshot("TextField property forwarded snapshot");
    });

    test("should match snapshot - TextField (default with empty theme)", () => {
      const component = renderer.create(<TextField {...getProps({ theme: {} })} />).toJSON();

      expect(component).to.matchSnapshot("TextField default with empty theme snapshot");
    });
  });
});
