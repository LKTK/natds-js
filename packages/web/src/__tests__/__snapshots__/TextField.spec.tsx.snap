// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component rendering states should match snapshot - TextField (default with empty theme): TextField default with empty theme snapshot 1`] = `
<div
  className="sc-bdVaJa wTxLS"
>
  <label
    className="sc-bwzfXH WyYxe"
    disabled={false}
    htmlFor="field"
  >
    Label
  </label>
  <input
    className="sc-htpNat iuSOjT"
    disabled={false}
    id="field"
    placeholder="Placeholder"
    type="text"
  />
  <span
    className="sc-bxivhb eyOWtv"
    disabled={false}
  >
    Assistive text
  </span>
</div>
`;

exports[`TextField component rendering states should match snapshot - TextField (default): TextField default snapshot 1`] = `
<div
  className="sc-bdVaJa wTxLS"
>
  <label
    className="sc-bwzfXH gNcVzU"
    disabled={false}
    htmlFor="field"
  >
    Label
  </label>
  <input
    className="sc-htpNat iXoZtg"
    disabled={false}
    id="field"
    placeholder="Placeholder"
    type="text"
  />
  <span
    className="sc-bxivhb bmtaJq"
    disabled={false}
  >
    Assistive text
  </span>
</div>
`;

exports[`TextField component rendering states should match snapshot - TextField (disabled): TextField disabled snapshot 1`] = `
<div
  className="sc-bdVaJa wTxLS"
>
  <label
    className="sc-bwzfXH fCrgGn"
    disabled={true}
    htmlFor="field"
  >
    Label
  </label>
  <input
    className="sc-htpNat JEDVS"
    disabled={true}
    id="field"
    placeholder="Placeholder"
    type="text"
  />
  <span
    className="sc-bxivhb eVoknC"
    disabled={true}
  >
    Assistive text
  </span>
</div>
`;

exports[`TextField component rendering states should match snapshot - TextField (disabled): TextField disabled snapshot 2`] = `
<div
  className="sc-bdVaJa wTxLS"
>
  <label
    className="sc-bwzfXH fCrgGn"
    disabled={true}
    htmlFor="field"
  >
    Label
  </label>
  <input
    className="sc-htpNat JEDVS"
    disabled={true}
    id="field"
    placeholder="Placeholder"
    type="text"
  />
  <span
    className="sc-bxivhb eVoknC"
    disabled={true}
  >
    Assistive text
  </span>
</div>
`;

exports[`TextField component rendering states should match snapshot - TextField (error): TextField error snapshot 1`] = `
<div
  className="sc-bdVaJa wTxLS"
>
  <label
    className="sc-bwzfXH bkyNlq"
    disabled={false}
    htmlFor="field"
  >
    Label
  </label>
  <input
    className="sc-htpNat QdZdV"
    disabled={false}
    id="field"
    placeholder="Placeholder"
    type="text"
  />
  <span
    className="sc-bxivhb hOxVCF"
    disabled={false}
  >
    Assistive text
  </span>
</div>
`;

exports[`TextField component rendering states should match snapshot - TextField (property forwarded): TextField property forwarded snapshot 1`] = `
<div
  className="sc-bdVaJa wTxLS"
>
  <label
    className="sc-bwzfXH gNcVzU"
    disabled={false}
    htmlFor="field"
  >
    Label
  </label>
  <input
    className="sc-htpNat iXoZtg"
    disabled={false}
    id="field"
    placeholder="Placeholder"
    type="password"
  />
  <span
    className="sc-bxivhb bmtaJq"
    disabled={false}
  >
    Assistive text
  </span>
</div>
`;

exports[`TextField component rendering states should match snapshot - TextField (required): TextField required snapshot 1`] = `
<div
  className="sc-bdVaJa wTxLS"
>
  <label
    className="sc-bwzfXH gNcVzU"
    disabled={false}
    htmlFor="field"
  >
    Label *
  </label>
  <input
    className="sc-htpNat iXoZtg"
    disabled={false}
    id="field"
    placeholder="Placeholder"
    type="text"
  />
  <span
    className="sc-bxivhb bmtaJq"
    disabled={false}
  >
    Assistive text
  </span>
</div>
`;

exports[`TextField component rendering states should match snapshot - TextField (success): TextField success snapshot 1`] = `
<div
  className="sc-bdVaJa wTxLS"
>
  <label
    className="sc-bwzfXH gNwsWk"
    disabled={false}
    htmlFor="field"
  >
    Label
  </label>
  <input
    className="sc-htpNat gLyFyr"
    disabled={false}
    id="field"
    placeholder="Placeholder"
    type="text"
  />
  <span
    className="sc-bxivhb jGNqxT"
    disabled={false}
  >
    Assistive text
  </span>
</div>
`;

exports[`TextField component rendering states should match snapshot - TextField (theme change): TextField theme change snapshot 1`] = `
<div
  className="sc-bdVaJa wTxLS"
>
  <label
    className="sc-bwzfXH gNcVzU"
    disabled={false}
    htmlFor="field"
  >
    Label
  </label>
  <input
    className="sc-htpNat iXoZtg"
    disabled={false}
    id="field"
    placeholder="Placeholder"
    type="text"
  />
  <span
    className="sc-bxivhb bmtaJq"
    disabled={false}
  >
    Assistive text
  </span>
</div>
`;
